app/lib/features/attachments/actions/handle_selected_attachments.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart'
+    show AttachmentDraft, AttachmentsManager;
+import 'package:flutter/material.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+import 'package:mime/mime.dart';
+
+final _log = Logger('a3::action::attachments');
+
+// if generic attachment, send via manager
+Future<void> handleAttachmentSelected({
+  required BuildContext context,
+  required WidgetRef ref,
+  required AttachmentsManager manager,
+  required List<File> attachments,
+  String? title,
+  String? link,
+  required AttachmentType attachmentType,
+}) async {
+  /// converts user selected media to attachment draft and sends state list.
+  /// only supports image/video/audio/file.
+  final lang = L10n.of(context);
+  EasyLoading.show(status: lang.sendingAttachment);
+  final client = ref.read(alwaysClientProvider);
+  List<AttachmentDraft> drafts = [];
+  try {
+    for (var selected in attachments) {
+      final file = selected;
+      final mimeType = lookupMimeType(file.path);
+      if (mimeType == null) throw lang.failedToDetectMimeType;
+      if (attachmentType == AttachmentType.camera ||
+          attachmentType == AttachmentType.image) {
+        Uint8List bytes = await file.readAsBytes();
+        final decodedImage = await decodeImageFromList(bytes);
+        final imageDraft = client
+            .imageDraft(file.path, mimeType)
+            .size(bytes.length)
+            .width(decodedImage.width)
+            .height(decodedImage.height);
+        final attachmentDraft = await manager.contentDraft(imageDraft);
+        drafts.add(attachmentDraft);
+      } else if (attachmentType == AttachmentType.audio) {
+        Uint8List bytes = await file.readAsBytes();
+        final audioDraft =
+            client.audioDraft(file.path, mimeType).size(bytes.length);
+        final attachmentDraft = await manager.contentDraft(audioDraft);
+        drafts.add(attachmentDraft);
+      } else if (attachmentType == AttachmentType.video) {
+        Uint8List bytes = await file.readAsBytes();
+        final videoDraft =
+            client.videoDraft(file.path, mimeType).size(bytes.length);
+        final attachmentDraft = await manager.contentDraft(videoDraft);
+        drafts.add(attachmentDraft);
+      } else {
+        String fileName = file.path.split('/').last;
+        final fileDraft = client
+            .fileDraft(file.path, mimeType)
+            .filename(fileName)
+            .size(file.lengthSync());
+        final attachmentDraft = await manager.contentDraft(fileDraft);
+        drafts.add(attachmentDraft);
+      }
+    }
+    if (attachmentType == AttachmentType.link && link != null) {
+      final attachmentDraft = await manager.linkDraft(link, title);
+      drafts.add(attachmentDraft);
+    }
```

Comment:
This is the only new addition in this function prior to earlier code.

Reply:
```
Future<void> handleAttachmentSelected({
  required BuildContext context,
  required WidgetRef ref,
  required AttachmentsManager manager,
  required List<File> attachments,
  String? title,
  String? link,
  required AttachmentType attachmentType,
})
```

This is the existing function I used to add all attachment [here](https://github.com/acterglobal/a3/pull/2048/files#r1715476655) with for loop.

In new, I used Pin State class where `PinAttachment` class I used for all attachment data including Link Type.

So for loop of `List<File> attachments, ` in `handleAttachmentSelected` doesn't match with new code.

I don't wanted to changes much in existing attachment module code so that's why handle it as it is but in later PRs I will optimised this function code.